.github/workflows/tests.yml

```diff
@@ -7,19 +7,68 @@ on:
   push:
     branches:
       - main
-      - v*.*.x
+      - v*.*
     tags:
       - v*.*.*
   pull_request:
 
 jobs:
+  build-cruby-gem:
+    name: "Build CRuby gem"
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v3
+        with:
+          path: ports/archives
+          key: archives-ubuntu-${{hashFiles('ext/re2/extconf.rb')}}
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.2"
+          bundler-cache: true
+      - run: ./scripts/test-gem-build gems ruby ${{github.ref_type}}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: cruby-gem
+          path: gems
+
+  build-precompiled-gems:
+    name: "Precompile ${{ matrix.platform }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - "aarch64-linux"
+          - "arm-linux"
+          - "arm64-darwin" # github actions does not support this runtime as of 2022-12, but let's build anyway
+          - "x64-mingw-ucrt"
+          - "x64-mingw32"
+          - "x86-linux"
+          - "x86-mingw32" # github actions does not support this runtime as of 2022-12, but let's build anyway
+          - "x86_64-darwin"
+          - "x86_64-linux"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v3
+        with:
+          path: ports/archives
+          key: archives-ubuntu-${{hashFiles('ext/re2/extconf.rb')}}
+      - run: |
+          docker run --rm -v "$(pwd):/re2" -w /re2 \
+            "ghcr.io/rake-compiler/rake-compiler-dock-image:1.3.0-mri-${{matrix.platform}}" \
+            ./scripts/test-gem-build gems ${{matrix.platform}} ${{github.ref_type}}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: "cruby-${{matrix.platform}}-gem"
+          path: gems
+
   compile-and-test-system-dependencies:
+    name: Ruby ${{ matrix.ruby }} - libre2.${{ matrix.libre2.soname }}
     needs: ["build-cruby-gem"]
-    name: System Dependencies - ${{ matrix.sys }} vendored libs - Ruby ${{ matrix.ruby }} - libre2 ABI version ${{ matrix.libre2.soname }}
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        sys: ["enable", "disable"]
```

Comment:
@stanhu do you have any thoughts about pulling this seeing as we have `compile-and-test-vendored-dependencies` which _should_ cover the same cases (compiling the gem and its vendored dependencies on all supported platforms)?

Reply:
As I recall, I added this in https://github.com/mudge/re2/pull/93 which covers the case when system libraries are installed, but the gem wants to use the vendored libraries and compile from source.

Does one of the checks in https://github.com/mudge/re2/actions/runs/6253089644/job/16977764503 cover this case?

Reply:
Ah, let me add a specific test case for that: using the vendored libraries but with libre2 installed in Ruby's `exec_prefix` (I guess we don't need to exhaustively test that against every platform and Ruby version?)

Reply:
I've added a dedicated job for testing the installation of the gem (with the default `---disable-system-libraries`) but where there is already a libre2 in the Ruby `exec_prefix` (and a system version of Abseil installed).

Reply:
I've updated this to use whatever `libre2-dev` is available on `ubuntu-latest` but still symlinking it into Ruby's `lib` directory.

Reply:
Sounds good.